src/devices/Nrf24l01/Nrf24l01.cs

```diff
@@ -0,0 +1,773 @@
+{
+    /// <summary>
+    /// Single chip 2.4 GHz Transceiver nRF24L01
+    /// </summary>
+    public class Nrf24l01 : IDisposable
+    {
+        private GpioController _gpio = null;
+        private SpiDevice _sensor = null;
+
+        private readonly int _ce;
+        private readonly int _irq;
+
+        private readonly byte[] _empty = new byte[0];
+
+        #region prop
+        private byte _packetSize;
+        /// <summary>
+        /// nRF24L01 Receive Packet Size
+        /// </summary>
+        public byte PacketSize
+        {
+            get { return _packetSize; }
+            set { _packetSize = value < 0 || value > 32 ? throw new ArgumentOutOfRangeException("PacketSize needs to be in the range of 0 to 32!") : value; }
+        }
+
+        /// <summary>
+        /// nRF24L01 Send Address
+        /// </summary>
+        public Span<byte> Address { get => ReadTxAddress(); set => SetTxAddress(value); }
+
+        /// <summary>
+        /// nRF24L01 Working Channel
+        /// </summary>
+        public byte Channel { get => ReadChannel(); set => SetChannel(value); }
+
+        /// <summary>
+        /// nRF24L01 Output Power
+        /// </summary>
+        public OutputPower OutputPower { get => ReadOutputPower(); set => SetOutputPower(value); }
+
+        /// <summary>
+        /// nRF24L01 Send Rate
+        /// </summary>
+        public DataRate DataRate { get => ReadDataRate(); set => SetDataRate(value); }
+
+        /// <summary>
+        /// nRF24L01 Power Mode
+        /// </summary>
+        public PowerMode PowerMode { get => ReadPowerMode(); set => SetPowerMode(value); }
+
+        /// <summary>
+        /// nRF24L01 Working Mode
+        /// </summary>
+        public WorkingMode WorkingMode { get => ReadWorkwingMode(); set => SetWorkingMode(value); }
+
+        /// <summary>
+        /// nRF24L01 Pipe 0
+        /// </summary>
+        public Nrf24l01Pipe Pipe0 { get; private set; }
+
+        /// <summary>
+        /// nRF24L01 Pipe 1
+        /// </summary>
+        public Nrf24l01Pipe Pipe1 { get; private set; }
+
+        /// <summary>
+        /// nRF24L01 Pipe 2
+        /// </summary>
+        public Nrf24l01Pipe Pipe2 { get; private set; }
+
+        /// <summary>
+        /// nRF24L01 Pipe 3
+        /// </summary>
+        public Nrf24l01Pipe Pipe3 { get; private set; }
+
+        /// <summary>
+        /// nRF24L01 Pipe 4
+        /// </summary>
+        public Nrf24l01Pipe Pipe4 { get; private set; }
+
+        /// <summary>
+        /// nRF24L01 Pipe 5
+        /// </summary>
+        public Nrf24l01Pipe Pipe5 { get; private set; }
+        #endregion
+
+        #region SpiSettings
+        /// <summary>
+        /// NRF24L01 SPI Clock Frequency
+        /// </summary>
+        public const int SpiClockFrequency = 2000000;
+
+        /// <summary>
+        /// NRF24L01 SPI Mode
+        /// </summary>
+        public const SpiMode SpiMode = System.Device.Spi.SpiMode.Mode0;
```

Comment:
don't think you need to specify System.Device.Spi namespace

Reply:
If I don't do this, VS will report an error